src/main.c

```diff
@@ -213,17 +213,26 @@ void c_main(void)
 	 * miss and therefore cause no evictions.
 	 */
 
+	int id = MMC_ID;
 	mmc_inited = !mmc_init(MMC_ID);
+	char *rootfs_dev = "root=/dev/mmcblk0p1";
+#ifdef MMC_SD
+	if (!mmc_inited) {
+		rootfs_dev = "root=/dev/mmcblk1p1";
```

Comment:
```"root=/dev/mmcblk" #MMC_ID2 "p1"```;

Reply:
I have mmcblk1, not mmcblk2 (as MMC_ID2 = 2)

Reply:
Right.

Reply:
Actually, you need a `#ifdef MMC_ID2` around this if block, or it will fail to build for the other configs (which don't define MMC_ID2).

Reply:
right, fixed